File: eclipse-workspace/basics/src/recursion/Palindrome.java
package recursion;

public class Palindrome {

	public static void main(String[] args) {
		int n = 151;
		int temp = 0;
		int pal = palindrome(n,temp);
		if(n==pal) System.out.println("It is a palindrome number");
		else System.out.println("It is not a palindrome number");
	}

	private static int palindrome(int n, int temp) {
		if(n==0) return temp;
		temp = (temp*10)+n%10;
		return palindrome(n/10,temp);
	}

}

File: Python/Programs/pythonProject/BasicExercise/palindrome.py
# check palindrome number
num = int(input('enter a number: '))
temp = num
rev = 0
while num>0:
    x = num%10
    rev = rev*10+x
    num = num//10
if temp == rev:
        print("it is a palindrome number")
else:
        print("it is not a palindrome number")


# check palindrome name
# name = "madam"
# if name[::-1] == name:
#     print("it is a palindrome word")
# else:
#     print("it is not a palindrome word")
